src/Adapter/Order/CommandHandler/UpdateProductInOrderHandler.php

```diff
@@ -95,36 +118,35 @@ public function handle(UpdateProductInOrderCommand $command)
                 throw new OrderException('An error occurred while editing the product line.');
             }
 
-            $cart = Cart::getCartByOrderId($order->id);
             if (!($cart instanceof Cart)) {
                 throw new OrderException('Cart linked to the order cannot be found.');
             }
             $product = $this->getProduct(new ProductId((int) $orderDetail->product_id), (int) $order->id_lang);
             $combination = $this->getCombination((int) $orderDetail->product_attribute_id);
 
-            // Add specific price for the product being added
-            $specificPrice = $this->createSpecificPriceIfNeeded(
+            $this->updateSpecificPrice(
                 $command->getPriceTaxIncluded(),
                 $command->getPriceTaxExcluded(),
                 $order,
-                $cart,
                 $product,
                 $combination
             );
 
-            if (null !== $specificPrice) {
-                $temporarySpecificPrices[] = $specificPrice;
```

Comment:
You can also remove the `$temporarySpecificPrices` variable which is not used any more